extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchConstantPostAggregator.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.druid.query.aggregation.datasketches.tuple;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.datasketches.tuple.arrayofdoubles.ArrayOfDoublesSketch;
+import org.apache.datasketches.tuple.arrayofdoubles.ArrayOfDoublesSketches;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.guava.Comparators;
+import org.apache.druid.query.aggregation.AggregatorFactory;
+import org.apache.druid.query.aggregation.AggregatorUtil;
+import org.apache.druid.query.cache.CacheKeyBuilder;
+import org.apache.druid.segment.data.SafeWritableMemory;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This post-aggregator converts a given Base64 encoded string to an ArrayOfDoublesSketch.
+ * The input column contains name of post-aggregator output and base64 encoded input string.
+ * The output is a deserialized {@link ArrayOfDoublesSketch} .
+ */
+public class ArrayOfDoublesSketchConstantPostAggregator extends ArrayOfDoublesSketchPostAggregator

```

Comment:
I didn't quite understand the need for this `PostAggregator` - can you please provide any reference or example explaining this a bit more in detail?
Also, given that this post agg just converts a user given string to a sketch, is it necessary to make it a post aggregator and not a simple function?

Reply:
@rohangarg : Requirement here is to provide an external sketch to the Native Query, this external sketch is non-time series in nature, so it is not present in any of the dimensions/fields. External Sketch is added at post-aggregation step so that we could do set operations on the sketches( time series sketch output in aggregation step vs non time series data)

Reply:
@rohangarg : Let me know if that clarifies or more clarity is required.

Reply:
An `ExpressionPostAggregator` could have been used as well where the expression is a function to convert string to sketch. But then, we have similar post aggregators already. E.g. ArrayOfDoublesSketchToStringPostAggregator

Reply:
But then, those post aggregators make sense since they operate on a sketch which is typically the output of an aggregation. This external sketch is coming out of an aggregate function?

Reply:
The external sketch is generated by a backend service external to druid.

Reply:
Basically assume we have a time series data as an output from aggregator output.
Filter => country ='US' and region In(Florida, California). Based on this we have sketches created while ingesting data. With userId being the key of the sketch and metrics in Double sketch is the frequency of the user appearing in the time series Data.

External sketch = Tuplesketch of all the users who watches Cricket.- This is non time series data. 

Now we take want to intersect the external sketch with aggregator output in post-aggregator step.

@abhishekagarwal87 @rohangarg

Reply:
This is an example use case, where we want to obtain time series data of Cricket Users and `sum` total of the clicks they have made in the app screen pixel.


```
{
  "queryType": "groupBy",
  "dataSource": "ds1",
  "granularity": "hour",
  "dimensions": [],
  "filter": {
    "type": "and",
    "fields": [
      { "type": "selector", "dimension": "country", "value": "US" },
      { "type": "selector", "dimension": "type", "value": 1 }

      , {"type": "in","dimension": "region",  "values": ["FLORIDA","CALIFORNIA"]}  ]
  },
  "aggregations": [
    { "type" : "arrayOfDoublesSketch", "name" : "filtered_user_sketch",  "fieldName": "filtered_user_sketch", "nominalEntries": 4096},
  ],
    "postAggregations": [

    {
      "type": "arrayOfDoublesSketchToSumMetricsEstimate",
      "name": "cricket_users_count",
      "field":
      {
        "type": "arrayOfDoublesSketchSetOp",
        "name": "filtered_cricket_users_tuple_sketch",
        "operation": "INTERSECT",
        "fields": [
        {

                        "type"  : "arrayOfDoublesSketchConstant",
                        "name"  : "cricket_users_tuple_sketch",
                        "value" : "AQEJAwgBzJP/////////fwIAAAAAAAAAzT6NGdX0aWUOJvS5EIhpLwAAAAAAAAAAAAAAAAAAAAA="


          },
          {
                        "type"  : "fieldAccess",
                        "fieldName"  : "filtered_user_sketch"
          }
        ],
        "nominalEntries":4096
      }
    }
  ],
  "intervals": ["2022-11-01/2023-02-01"]
}



```

Reply:
Thanks a lot for explaining the use-case @anshu-makkar - it was very helpful!
Ideally, I think it would have been better if we could just plug a deserialization function and make `ExpressionPostAggregator` work with it as @abhishekagarwal87 had mentioned.
But I think it is ok to have a dedicated post agg for this now. Once there is a consolidate method of combining serdes with post aggregations, we can consider deprecating some of these post aggregations.